### PR TITLE
Iodide 1899 remove global for process env

### PR DESCRIPTION
**Pull request by Algentile**
Creating pull Request to migrate from method of parsing environment into IODIDE_BUILD_MODE variable and instead using webpack's defined process.env.NODE_ENV variable. This is a code quality issue in order to align the project more towards Webpack best practices. (#1899).

### Pull Request checklist

- [N / A] **Documentation**: If this feature has or requires documentation, the relevant docs have been updated.
- [ ] **Changelog**: This PR updates the [changelog](../CHANGELOG.md) with any user-visible 
changes.
Changelog has been modified to describe change that was made. 
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
There are no automated tests as there were only two major cases that had to be verified via the command line. 
First Case: Is process.env.NODE_ENV exposed properly to the react components on build (pass)
Seconds case: Is process.env.NODE_ENV exposed to webpack.config.js on build. (pass)

### CODE REVIEW

**Comment by bcolloran** [comment]
Hi @Algentile thanks so much for the PR! I just wanted to send a quick note to say that our team is at a company meeting this week, and that we may not get to this PR for a while -- just didn't want to leave you hanging! :-) We'll try to review this as soon as we're able. Thanks!

**Comment by bcolloran** [comment]
@Algentile thanks for getting the ball rolling on this! I've added a couple more commits on top of your work and rebased on the latest master to get this branch in sync with the latest.

**Comment by bcolloran** [comment]
@wlach and @hamilton -- i think this is quite straight forward and should be a simple review. would one of you mind deploying to a test server to make sure it all still works in the non-local setting? thanks!

**Comment by wlach** [comment]
I'll try deploying this PR to iodide stage

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,8 @@
 - Gravatars are served via https by default (fixes #1930)
 - Add new file management UI (click _Menu > Manage Files_ while logged in and
   viewing a notebook that you own)
+- Migrate from parsing environment variable with global IODIDE_BUILD_MODE to
+  using webpack's process.env.NODE_ENV and the environmentPlugin code quality (#1899)
```
Comment (wlach) [inline CHANGELOG.md]: I am not sure if this PR needs a changelog entry, as there is nothing user-visible that's different

**Comment by bcolloran** [comment]
sounds good @wlach -- i'll remove the changelog entry and merge. Thanks for reviewing! and thanks again @Algentile for the PR!

**Comment by Algentile** [comment]
Thanks for reviewing @wlach, and thanks @bcolloran for migrating the other variables to using process.env.

**Comment by wlach** [comment]
@Algentile if you are interested in working on more things in iodide, feel free to ping us on [gitter](https://gitter.im/iodide-project/iodide).

**Comment by Algentile** [comment]
@wlach Thanks I will hop on, I am relatively new to the project and definitely willing to help whenever I can.

**Pull request merged**